channels/channel.streams/slam/chn_slam.py

```diff
@@ -0,0 +1,200 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import datetime
+
+from resources.lib import chn_class, mediatype
+from resources.lib.helpers.htmlentityhelper import HtmlEntityHelper
+
+from resources.lib.mediaitem import MediaItem
+from resources.lib.logger import Logger
+from resources.lib.helpers.jsonhelper import JsonHelper
+from resources.lib.urihandler import UriHandler
+from resources.lib.parserdata import ParserData
+from resources.lib.streams.m3u8 import M3u8
+
+
+class Channel(chn_class.Channel):
+    """
+    main class from which all channels inherit
+    """
+
+    def __init__(self, channel_info):
+        """Initialisation of the class.
+
+        Arguments:
+        channel_info: ChannelInfo - The channel info object to base this channel on.
+
+        All class variables should be instantiated here and this method should not
+        be overridden by any derived classes.
+
+        """
+
+        chn_class.Channel.__init__(self, channel_info)
+
+        # ==== Actual channel setup STARTS here and should be overwritten from derived classes =====
+        self.noImage = ""
+
+        # setup the urls
+        if self.channelCode == 'slam':
+            self.noImage = "slam.png"
+            self.mainListUri = "https://api.slam.nl/api/getStreams?brand=slam"
+            self.baseUrl = "http://www.slam.nl"
+
+        # setup the main parsing data
+        self._add_data_parser(self.mainListUri, match_type=ParserData.MatchExact, json=True,
+                              preprocessor=self.add_live_streams)
```

Comment:
Now get rid of the complete `self.add_live_streams` method and content. And add a parser for "self.mainListUri" that contains a parser `["data", "streams"]` and a creator `self.create_live_stream` and set `json=True`. 

The in the new method `def create_live_stream(result_set)` build a `MediaItem` from the input. That input will look like a dictionary with this content: 

```
      {
        "id": "2",
        "type": "",
        "orderNum": "1",
        "startDate": "2018-12-31 14:03:00",
        "endDate": "2029-12-31 00:00:00",
        "brandId": "2",
        "title": "SLAM!",
        "shortName": "",
        "streamImage": "1558361412_SLAM_.jpg",
        "playerImage": "1551091620_Slam_stream2.png",
        "streamUrl": "slam",
        "streamMetaUrl": "https://live.slam.nl/metadata/livewall",
        "livestreamEmbedUrl": "",
        "freeWheelTag": "",
        "freeWheelTags": {
          "smarttv": "sbs_radiocorp_slamfm_smarttv",
          "ios_mobile_web": "sbs_radiocorp_slamfm_ios_mobile_web",
          "ios_app": "sbs_radiocorp_slamfm_ios_app",
          "ios_airplay": "sbs_radiocorp_slamfm_ios_airplay",
          "desktop": "sbs_radiocorp_slamfm_desktop",
          "chromecast": "sbs_radiocorp_slamfm_chromecast",
          "android_mobile_web": "sbs_radiocorp_slamfm_android_mobile_web",
          "android_app": "sbs_radiocorp_slamfm_android_app"
        },
        "metadata": {
          "nowArtist": "Martin Jensen",
          "nowTitle": "Solo Dance",
          "nowDuration": "170"
        }
      }
```

And with that you can create  a nice MediaItem object with an image and perhaps the current song? Make sure you set the `.isLive=True` on the MediaItem. 

The URL will look like: `https://playerservices.streamtheworld.com/api/livestream?station=slam&transports=http%2Chls%2Chlsts&version=1.9&request.preventCache=1646864848160`  Where the last number should be random. 

The create a DataParser that only updates for the url `https://playerservices.streamtheworld.com/api/livestream` and that parser, loads the content from the url, construct a stream and sets the correct url in 

```
        stream = item.add_stream(stream_url, 0)
        M3u8.set_input_stream_addon_input(stream, self.headers)
        item.complete = True
```